packages/pdfform/src/fillPdf.ts

```diff
@@ -1,43 +1,44 @@
-import {spawn} from 'child_process';
-// @ts-ignore
-import pdftk from 'node-pdftk';
-import type {Readable} from 'stream';
-import generateFdfFromJson from './generateFdfFromJson';
+import {Readable} from 'stream';
+import {PDFDocument} from 'pdf-lib';
+import fs from 'fs';
+
+async function fillPdf(sourcePdfFilename: string, data: {[key: string]: string | undefined}): Promise<Uint8Array> {
+	return new Promise(resolve => {
+		fs.readFile(sourcePdfFilename, async (error, file) => {
+			if (error) throw error;
+			const pdfDoc = await PDFDocument.load(file);
+			const form = pdfDoc.getForm();
+			Object.keys(data).forEach(key => {
+				form.getTextField(key).setText(data[key]);
+			});
+			const pdfBytes = await pdfDoc.save();
+			resolve(pdfBytes);
+		});
+	});
+}
 
 /**
  * Take mapped pdf form data and fill a PDF form, returning a read stream
  * @param sourcePdfFilename
  * @param data
  */
 export async function pdfForStream(sourcePdfFilename: string, data: {[key: string]: string | undefined}): Promise<Readable> {
-	// implement taking a source PDF filename and mapped data from mapPdfFormData and returning a PDF stream
-	return new Promise((resolve, reject) => {
-		const fdfData = generateFdfFromJson(data);
-		const options = [sourcePdfFilename, 'fill_form', '-', 'output', '-', 'flatten'];
-
-		const child = spawn('pdftk', options);
-
-		child.stdin.write(fdfData);
-		child.on('error', reject);
-		child.stdin.on('error', reject);
-		child.stdin.on('close', () => resolve(child.stdout));
-		child.stdin.end();
-	});
+	return Readable.from(await fillPdf(sourcePdfFilename, data));
 }
 
 /**
  * Take mapped pdf form data and fill a PDF form, returning a buffer
  * @param sourcePdfFilename
  * @param data
  */
-export function pdfForBuffer(sourcePdfFilename: string, data: {[key: string]: string | undefined}): Buffer {
-	return pdftk.input(sourcePdfFilename).fillForm(data).flatten().output();
+export async function pdfForBuffer(sourcePdfFilename: string, data: {[key: string]: string | undefined}): Promise<Buffer> {
```

Comment:
When making a function async, you need to check where this is all used. The calling code may not be async.